src/daft-local-execution/src/run.rs

```diff
@@ -92,6 +124,9 @@ impl NativeExecutor {
                 results_buffer_size,
             )
         })?;
+
+        let out = BlockingStreamIter::new(out);
```

Comment:
can you add a comment on why we need to block here?

Reply:
I am not completely sure but aren't we not leveraging python async really? → we are just blocking for a task to complete